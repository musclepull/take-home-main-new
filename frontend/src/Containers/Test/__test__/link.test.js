import React from "react"
import ReactDom  from "react-dom"
import TestLink from "../link"
import renderer from 'react-test-renderer'

it("renders without crashing", () => {
    const div = document.createElement("div")
    ReactDom.render(<TestLink/>, div)
})

test('Button changes the class when clicked', () => {
    const component = renderer.create(
      <TestLink>Facebook</TestLink>,
    )
    let tree = component.toJSON()
    expect(tree).toMatchSnapshot()
  
    // manually trigger the callback
    tree.props.onMouseEnter()
    // re-rendering
    tree = component.toJSON()
    expect(tree).toMatchSnapshot()

    // manually trigger the callback
    tree.props.onMouseLeave()
    // re-rendering
    tree = component.toJSON()
    expect(tree).toMatchSnapshot()
  })