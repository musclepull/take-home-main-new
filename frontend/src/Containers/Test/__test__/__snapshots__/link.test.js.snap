// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Button changes the class when clicked 1`] = `
<a
  className="normal"
  href="#"
  onMouseEnter={[Function]}
  onMouseLeave={[Function]}
>
  Facebook
</a>
`;

exports[`Button changes the class when clicked 2`] = `
<a
  className="hovered"
  href="#"
  onMouseEnter={[Function]}
  onMouseLeave={[Function]}
>
  Facebook
</a>
`;

exports[`Button changes the class when clicked 3`] = `
<a
  className="normal"
  href="#"
  onMouseEnter={[Function]}
  onMouseLeave={[Function]}
>
  Facebook
</a>
`;
